### PR TITLE
Allow undef for logout_uri and logout_redirect

### PR DESCRIPTION
**Pull request by abujeda**
`logout_uri` and `logout_redirect` default values are managed in ood-portal application.
Without the Optional[String] settings for these properties, it is not possible to remove the logout redirect from Apache.

There are cases, when multiple domains and OIDC are used, that a more complex logout redirect is required.
This can be now accomplished using the new `custom_vhost_directives` like so:
```yaml
openondemand::custom_vhost_directives:
  - 'RewriteRule ^/logout$ /oidc?logout=https://%{literal("%")}{HTTP_HOST} [R=301,L]'
```

This is simply propagating the current domain to the OIDC identity provider for any domain used within OOD.

Although the `Redirect` directive created with the default values for `logout_uri` and `logout_redirect` do not affect the functionality of the `RewriteRule` as the `custom_vhost_directives` are printed first, it will create a cleaner config.

### CODE REVIEW

**Comment by abujeda** [comment]
relates to [#3442](https://github.com/OSC/ondemand/issues/3442)

**Comment by treydock** [comment]
You only need to change the datatype.  You don't need to remove the defaults.  Linters won't let `Optional` be used without default being `undef` I believe:

```
Variant[String[1], Undef]
```

Could try this too but I think linters will complain.

```
Optional[String[1]]
```

In Hiera you could set:

```
openondemand::logout_uri: ~
```

So you can force undef without removing the default from this module.

The reason I don't want to remove the defaults is I try to keep this module to have same defaults as OnDemand which helps with maintenance.

File: manifests/init.pp [hunk 1]
```diff
@@ -296,8 +296,8 @@
   Optional[Struct[{ url => String, id => String }]] $analytics = undef,
   String $public_uri = '/public',
   String $public_root = '/var/www/ood/public',
-  String $logout_uri = '/logout',
-  String $logout_redirect = '/pun/sys/dashboard/logout',
+  Optional[String] $logout_uri = undef,
+  Optional[String] $logout_redirect = undef,
```
[the remark this answers is not in the document]
Comment (abujeda) [reply manifests/init.pp]: Thanks - I have updated with the changes to the `REFERENCE.md `too

**Comment by treydock** [comment]
This is released as 5.0.1.

**Pull request merged**